hls/lib/hls_endpoint.ex

```diff
@@ -91,10 +91,27 @@ defmodule Membrane.RTC.Engine.Endpoint.HLS do
                 description: """
                 HLS stream and playlist configuration.
                 """
+              ],
+              subscribe_mode: [
+                spec: :auto | :manual,
+                default: :auto,
+                description: """
+                Whether tracks should be subscribed automatically when they're ready.
+                If set to `:manual` hls endpoint will subscribe only to tracks send using message:
+                `{:subscribe, tracks}`
+                """
               ]
 
   @impl true
   def handle_init(_context, options) do
+    unless options.subscribe_mode in [:auto, :manual] do
+      raise """
+      Cannot initialize HLS endpoint.
+      Invalid value for `:subscribe_mode`: #{options.subscribe_mode}.
+      Please set `:subscribe_mode` to either `:auto` or `:manual`.
+      """
+    end
```

Comment:
What do you think about that?
```suggestion
  @impl true
  def handle_init(_context, options) when options.subscribe_mode not in [:auto, :manual], do: raise """
      Cannot initialize HLS endpoint.
      Invalid value for `:subscribe_mode`: #{options.subscribe_mode}.
      Please set `:subscribe_mode` to either `:auto` or `:manual`.
      """

  @impl true
  def handle_init(_context, options) do
```